### PR TITLE
feat: Allow for changing cached reads inside of BlockBuildingHelperFromDB

### PR DESCRIPTION
**Pull request by 7suyash7**
#178 

- Added `update_cached_reads` method to the `BlockBuildingHelper` trait
- Implemented `update_cached_reads` for `BlockBuildingHelperFromDB`
- Updated `MockBlockBuildingHelper` to include an unimplemented `update_cached_reads` method

## ✅ I have completed the following steps:

* [ ✅] Run `make lint`
* [✅ ] Run `make test`

### CODE REVIEW

File: crates/rbuilder/src/building/builders/block_building_helper.rs [hunk 1]
```diff
@@ -256,6 +259,11 @@ impl<DB: Database + Clone + 'static> BlockBuildingHelperFromDB<DB> {
         self.built_block_trace.true_bid_value = true_value;
         Ok(())
     }
+
+    /// Updates the cached reads for the block state.
+    pub fn update_cached_reads(&mut self, cached_reads: CachedReads) {
+        self.block_state = self.block_state.clone().with_cached_reads(cached_reads);
+    }
```
Comment (ZanCorDX) [inline crates/rbuilder/src/building/builders/block_building_helper.rs]: Why is this here as a pub function and not implemented directly on impl BlockBuildingHelper for BlockBuildingHelperFromDB?

Comment (7suyash7) [reply crates/rbuilder/src/building/builders/block_building_helper.rs]: Sorry, my bad, just fixed it.

**Review by ZanCorDX — APPROVED** [review]
